xmodule/docs/decisions/0003-library-content-block-schema.rst

```diff
@@ -0,0 +1,201 @@
+
+* Next, we want to add support for non-randomized, hand-picked reference of library blocks. This is is a highly requested feature across the community.
+
+* In the future, we would like to support other modes: in particular, we want authors to eventually be able to hand-pick a set of blocks from a library, and *then* have the LMS randomly select a specified number of blokcs for each learner.
+
+We need to choose how to evolve the schema of the library_content block to support all of these incremental improvements. This is the current schema:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Field
+     - Scope, Type
+     - Default
+     - Example
+     - Description
+
+   * - source_library_id
+     - settings, str|None
+     - None
+     - "library-v1:OEX+Lib1"
+     - Key of V1 source library; None means unselected.
+
+   * - source_library\_ version
+     - settings, str|None
+     - None
+     - TBD
+     - Version of V1 source library (a MongoID string). None means latest.
+
+   * - mode
+     - settings, str
+     - "random"
+     - "random"
+     - How blocks should be selected. Only valid value is "random".
+
+   * - capa_type
+     - settings, str
+     - "any"
+     - "symbolicresponse"
+     - Unless "any": Include only problems, and only problems of this response type.
+
+   * - children
+     - special, list of XBlocks
+     - []
+     - ``block-v1:...+type@problem+block@A``
+       ``block-v1:...+type@problem+block@B``
+       ``block-v1:...+type@problem+block@D``
+       ``block-v1:...+type@problem+block@F``
+     - Copy of library contents, filtered by capa_type. Course author may apply edits that are local to this library_content block usage. Edits to child blocks' settings persist across library version upgrades. Edits to child blocks' content are wiped out when library version is upgraded.
+
+   * - max_count
+     - settings, int
+     - 1
+     - 2
+     - Number of children to randomly select for each learner. -1 means all.
+
+   * - selected
+     - user_state, list of (block type, block id) tuples
+     - []
+     - [("problem", "A"), ("problem", "F")]
+     - An ordered sub-list of the children, to be shown to a given learner. Once populated, this is kept stable for the learner, unless a the children or max_count are changed, in which case it will have blocks added or removed as necessary. If the list changes, it is re-shuffled.
+
+
+Decision
+********
+
+We will introduce the support for a "manual selection mode", without explicitly making it *mode* as hinted at by the old block schema. This will lead to a cleaner and more flexible implementation. We will achieve this with two new boolean settings: **manual** and **shuffle**:
```

Comment:
Do you think a boolean shuffle is sufficient to cover our use cases? There's nothing on the horizon that would warrant a more open-ended `mode` like enumeration for the selection criteria?

Reply:
Oh nvm, I see the section in "rejected alternatives"